pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -158,9 +160,29 @@ def read(self) -> DataFrame:
             parse_options=pyarrow_csv.ParseOptions(**self.parse_options),
             convert_options=pyarrow_csv.ConvertOptions(**self.convert_options),
         )
-        if self.kwds["dtype_backend"] == "pyarrow":
+
+        dtype_backend = self.kwds["dtype_backend"]
+
+        # Convert all pa.null() cols -> float64 (non nullable)
+        # else Int64 (nullable case)
```

Comment:
This was really confusing for me, but apparently even ``convert_dtypes`` on an all null column(float64) will change  it to an Int64.

Reply:
I am not sure we should replicate that here. 
In `convert_dtypes`, I suppose that is the consequence of trying to cast a float column where all non-null values can faithfully be represented as an integer. Which gives this corner case where this set of values is empty. 

But if you don't know the dtype you are starting with (like here), I think using float is safer (it won't give problems later on if you then want to use it for floats and not just integers). Or actually using object dtype might even be safest (as that can represent everything), but of course we don't (yet) have a nullable object dtype, so that's probably not really an option.

Reply:
The problem is that the C and python parsers already convert to an Int64. Is this something that we need to fix before 2.0?
```
>>> import pandas as pd
>>> from io import StringIO
>>> f = StringIO("a\n1,")
>>> pd.read_csv(f, dtype_backend="numpy_nullable")
      a
1  <NA>
>>> f.seek(0)
0
>>> pd.read_csv(f, dtype_backend="numpy_nullable").dtypes
a    Int64
dtype: object
>>> f.seek(0)
0
>>> pd.read_csv(f)
    a
1 NaN
>>> f.seek(0)
0
>>> pd.read_csv(f).dtypes
a    float64
dtype: object
```

Reply:
cc @phofl.

Reply:
We do this everywhere, so I am +1 on doing Int64 here as well. Initially this came from read_parquet I think